File: ruby/b_polygon.rb
# https://atcoder.jp/contests/abc117/tasks/abc117_b
def main
  _n = gets.to_i # unused
  l = gets.split.map(&:to_i)

  max = l.max

  rem = l.dup
  rem.delete_at(l.index(max))
  rem_sum = rem.sum

  result = drawable?(max, rem_sum) ? 'Yes' : 'No'
  puts result
end

# @param max [Integer] 一番長い辺
# @param rem_sum [Integer] 残りの辺の合計
# @return [Boolean]
def drawable?(max, rem_sum)
  max < rem_sum
end

main

File: cpp/b_polygon.cpp
#include <iostream>
using namespace std;

bool is_drawable(int max, int rem_sum)
{
  if (max < rem_sum)
  {
    return true;
  }
  return false;
}

int main()
{
  string result;
  int n;
  cin >> n;

  int maxL = 0, sum = 0;

  for (int i = 0; i < n; i++)
  {
    int L;
    cin >> L;
    sum += L;
    maxL = max(maxL, L);
  }

  int rem_sum = sum - maxL;

  if (is_drawable(maxL, rem_sum))
  {
    result = "Yes";
  }
  else
  {
    result = "No";
  }

  cout << result << endl;
}
